src/GravitationalClustering.js

```diff
@@ -0,0 +1,189 @@
+'use strict';
+
+var Particle = require('./Particle');
+var UnionFind = require('ml-disjoint-set');
+var Distance = require('ml-distance').distance;
+var randomInt = require('./Utils').randomInt;
+
+/**
+ * @class ParticleCluster
+ */
+class ParticleCluster {
+    /**
+     * ParticleCluster constructor.
+     * @param particle {Particle} - Initial particle for the cluster (optional)
+     */
+    constructor(particle) {
+        this.elements = [];
+        if (particle !== undefined) {
+            this.elements.push(particle);
+        }
+    }
+}
+
+/**
+ * @class GravitationalClustering
+ */
+class GravitationalClustering {
+
+    /**
+     * Create a new instance of a Gravitational Clustering algorithm
+     * @param {Object} options
+     * @param {Boolean} [options.unitMass] - the mass of each point are equal.
+     * @param {Number} [options.GC] - Gravitational constant (default: 10^-4)
+     * @param {Number} [options.deltaGC] - Constant to reduce the gravitational constant (default: 0.001)
+     * @param {Number} [options.alpha] - Minimum percentage of the dataset to become a cluster (default: 0.03)
+     * @param {Number} [options.gamma] - Percentage of space used as the radius of each particle used to create the clusters (default: 0.2)
+     * @param {Number} [options.dist] - Distance function that takes two points, see [ml-distance]{@link https://github.com/mljs/distance}
+     * @param X {Array} - Points to create the clusters (Optional)
+     * @param masses - Mass of each point for clustering (Optional)
+     */
+    constructor(options, X, masses) {
+        if (options === undefined) options = {};
+        this.unitMass = options.unitMass ? options.unitMass : true;
+        this.GC = options.GC ? options.GC : Math.pow(10, -4);
```

Comment:
Why don't call this as `gravitationalConstant`? it's not clear the name, the same for delta